packages/catalog-generator/src/main/java/io/kaoto/camelcatalog/commands/GenerateCommand.java

```diff
@@ -0,0 +1,99 @@
+        library.setName(configBean.getCatalogsName());
+
+        File outputFolder = createSubFolder(configBean.getOutputFolder());
+        cleanFolder(outputFolder);
+
+        configBean.getCatalogVersionSet()
+                .forEach(catalogCliArg -> {
+                    String runtimeFolderName = "camel-" + catalogCliArg.getRuntime().name().toLowerCase();
+                    File runtimeFolder = createSubFolder(outputFolder, runtimeFolderName);
+                    File catalogDefinitionFolder = createSubFolder(runtimeFolder, catalogCliArg.getCatalogVersion());
+
+                    LOGGER.info("Generating catalog: " + catalogCliArg.getRuntime() + " "
+                            + catalogCliArg.getCatalogVersion() + "\n");
+
+                    CatalogGeneratorBuilder builder = new CatalogGeneratorBuilder();
+                    var catalogGenerator = builder.withRuntime(catalogCliArg.getRuntime())
+                            .withCamelCatalogVersion(catalogCliArg.getCatalogVersion())
+                            .withKameletsVersion(configBean.getKameletsVersion())
+                            .withCamelKCRDsVersion("2.3.1")
+                            .withOutputDirectory(catalogDefinitionFolder)
+                            .build();
+
+                    CatalogDefinition catalogDefinition = catalogGenerator.generate();
+                    File indexFile = catalogDefinitionFolder.toPath().resolve(catalogDefinition.getFileName()).toFile();
+                    File relateIndexFile = outputFolder.toPath().relativize(indexFile.toPath()).toFile();
+
+                    catalogDefinition.setFileName(relateIndexFile.toString());
+
+                    library.addDefinition(catalogDefinition);
+                });
+
+        ObjectMapper jsonMapper = new ObjectMapper()
+                .configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+
+        var indexFile = outputFolder.toPath().resolve("index.json").toFile();
+        try {
+            jsonMapper.writerWithDefaultPrettyPrinter().writeValue(indexFile, library);
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing index file", e);
+        }
+
+    }
+
+    private File createSubFolder(File parentFolder, String folderName) {
+        File newSubFolder = parentFolder.toPath().resolve(folderName).toFile();
+        if (!newSubFolder.exists()) {
+            newSubFolder.mkdirs();
+        }
+        return newSubFolder;
+    }
+
+    private File createSubFolder(File parentFolder) {
+        File newSubFolder = parentFolder.toPath().toFile();
+        if (!newSubFolder.exists()) {
+            newSubFolder.mkdirs();
+        }
+        return newSubFolder;
+    }
+
+    private File cleanFolder(File folder) {
+        File[] files = folder.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                file.delete();
+            }
+        }
+
+        return folder;
```

Comment:
to double check but I think i've seen that you have Apache commons io as dependency, in this case you shoudlbe able to use FileUtils.deleteDirectory(dir);

Reply:
Oh, I didn't know, I'll update it :ok_hand: